node/src/components/storage.rs

```diff
@@ -684,7 +775,7 @@ impl Storage {
                         .insert(*block_hash, execution_result);
                     let was_written =
                         txn.put_value(self.deploy_metadata_db, &deploy_hash, &metadata, true)?;
-                    assert!(
+                    debug_assert!(
```

Comment:
This will now fail completely silently in production. We should probably at least log an error? (Also below.)
But even then: Is it okay to just continue if we failed to write to storage?

Reply:
It depends on the failure. In general, if we can continue to function with some loss of functionality, we should.

In this case, the execution results have already been applied to the Merkle trie in the `ContractRuntime`. The system can continue to make blocks with the rest of the chain. The loss of functionality appears to be that certain queries by the JSON-RPC may fail.

Reply:
I have added logging in this PR: https://github.com/casper-network/casper-node/pull/1711/commits/ac69ed9fda3315ff4bcab237adc6eca0509008d2